src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,54 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private Node<T> tail;
+    private Node<T> head;
+    private int size;
 
+    public void push(T value) {
+        Node<T> node = new Node<>(value, null);
+        if (head == null) {
+            head = node;
+        } else {
+            node.previous = tail;
+        }
+        tail = node;
+        size++;
```

Comment:
If you take the previous point into account, this method can be written in a couple of lines of code.

Reply:
done